app/react/V2/Routes/Settings/Filters/components/FiltersSidepanel.tsx

```diff
@@ -60,19 +60,21 @@ const FiltersSidepanel = ({
 
   const closeSidepanel = () => {
     setShowSidepanel(false);
+    reset();
   };
 
   const formatSelected = (selected: string[] | undefined) =>
     selected?.map(selection => {
       const templateName = allTemplates?.find(template => template._id === selection)?.name;
-      return { id: selection, name: templateName };
+      return { id: selection, name: templateName, rowId: selection };
     });
 
-  const handleSave = (values: Omit<ClientSettingsFilterSchema, 'items'> & { items: string[] }) => {
-    const result = { ...values, items: formatSelected(values.items) };
+  const handleSave = (values: Omit<Filter, 'subRows'> & { subRows: string[] }) => {
+    const result = { ...values, subRows: formatSelected(values.subRows) };
 
     if (!filter?._id) delete result._id;
     if (!filter?.id) result.id = uniqueID();
+    if (result.rowId === defaultValues.rowId) result.rowId = uniqueID();
```

Comment:
To avoid this comparison, uniqueID() could be assigned directly instead of the value 'NEW'

Reply:
Calling the function for uniqueID on the instantiation of the form was triggering a rerender loop. I had a hard time figuring out was was triggering the endless rerenders on the component and removing the uniqueID from the default values fixed it.

Reply:
I got it, it doesn't matter 👍